# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/librsvg/hotfix-rustc-t2.patch
# Copyright (C) 2022 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- librsvg-2.54.5/vendor/cfg-expr/src/targets.rs.vanilla	2022-09-01 13:44:22.215000000 +0200
+++ librsvg-2.54.5/vendor/cfg-expr/src/targets.rs	2022-09-01 14:03:53.911000000 +0200
@@ -353,8 +353,9 @@
 /// assert!(cfg_expr::targets::get_builtin_target_by_triple("x86_64-unknown-linux-musl").is_some());
 /// ```
 pub fn get_builtin_target_by_triple(triple: &str) -> Option<&'static TargetInfo> {
+    let trip = str::replace(triple, "-t2-", "-unknown-");
     ALL_BUILTINS
-        .binary_search_by(|ti| ti.triple.as_ref().cmp(triple))
+        .binary_search_by(|ti| ti.triple.as_ref().cmp(&trip))
         .map(|i| &ALL_BUILTINS[i])
         .ok()
 }
--- librsvg-2.54.5/vendor/cfg-expr/.cargo-checksum.json.vanilla	2022-09-01 14:20:54.437000000 +0200
+++ librsvg-2.54.5/vendor/cfg-expr/.cargo-checksum.json	2022-09-01 14:21:05.696000000 +0200
@@ -1 +1 @@
-{"files":{"CHANGELOG.md":"b2d467dd6af2bedd644d9ab96c27b600295d736c8020dd4aafb33ca95bdea7a7","CODE_OF_CONDUCT.md":"ff8e424cc9ba55e10c0f83cda8bbd9034ffa6a299d589804292837aa7d50e098","CONTRIBUTING.md":"d6e246265d4ada5659a30e7c15969e65d740584b468327af3e8fec54b7206fb6","Cargo.lock":"21209cafe999f71cf78402b9ab4402d3268ceea57f51998ad4c100ea1d53d47d","Cargo.toml":"7f5681f1ff50b5f4b35aec091bbcb7b74b11d7227c4b5e6ac6886c4f7c56f7c8","LICENSE-APACHE":"8173d5c29b4f956d532781d2b86e4e30f83e6b7878dce18c919451d6ba707c90","LICENSE-MIT":"090a294a492ab2f41388252312a65cf2f0e423330b721a68c6665ac64766753b","README.md":"e934d909a97ddbde3bf7c1e9350895623c3c237fed3f33ccc10261e4517b3d49","clippy.toml":"cba24bf1199526b940d8fe1df76bc10a3c71b404969a702ff18e0b73d90680c0","deny.toml":"b7e686ed2f16b1c84a183dcc4fa9795c67be96a9f8e54db73b95ae58d4b76479","examples/eval.rs":"e50e4f4e6d9d0a774acde0b8dff8e2959f521091ba2d7429bab9dc81c1a5fd0a","release.toml":"42df748d28abd726fa155a29102da0c7ac006dfb601f6bdebfc169049b0ca535","src/error.rs":"f7b5ac813cdaee0d6d021619caff52710918365e13ceb40bbc15f2e526aaf61e","src/expr.rs":"944b8ab027c4dcaa2f7ff7f1df6f23582bad5810af55049ce43bf404700b3059","src/expr/lexer.rs":"95fcdf902827fcccbc94af6052550b5070bc167c1d9c0b9bd5fb598f2b3380a3","src/expr/parser.rs":"4c5864d46b8bb023bd45b4eb2579f0c591b61f5f88d1f16b8e1e0fd4d00095c5","src/lib.rs":"68e82be4c5b8e454fbb3c524b6cc8b9534950bc4f99d9340766eb75405cd9abe","src/targets.rs":"1ae2ebf1096c746da6be68eb9059f200a4433cd07e27748f22ffe812131ed543","src/targets/builtins.rs":"f078966ea9ec6f5b003664ad36a7598dadb11179188643ae1adceabbaf7893ab","tests/eval.rs":"0c623b4ecd0b0cd69a666954c967f151b35610c8d23ed3f419954f8fef3b2878","tests/lexer.rs":"55109f29f4f3e257e969b53ee7823aeb50212abebbc55d1a67d7393ea42ce80b","tests/parser.rs":"466f77b150fe39eb89db3a157f26431b9367e872630fe213be75f9ffd8177e03"},"package":"0aacacf4d96c24b2ad6eb8ee6df040e4f27b0d0b39a5710c30091baa830485db"}
\ No newline at end of file
+{"files":{"CHANGELOG.md":"b2d467dd6af2bedd644d9ab96c27b600295d736c8020dd4aafb33ca95bdea7a7","CODE_OF_CONDUCT.md":"ff8e424cc9ba55e10c0f83cda8bbd9034ffa6a299d589804292837aa7d50e098","CONTRIBUTING.md":"d6e246265d4ada5659a30e7c15969e65d740584b468327af3e8fec54b7206fb6","Cargo.lock":"21209cafe999f71cf78402b9ab4402d3268ceea57f51998ad4c100ea1d53d47d","Cargo.toml":"7f5681f1ff50b5f4b35aec091bbcb7b74b11d7227c4b5e6ac6886c4f7c56f7c8","LICENSE-APACHE":"8173d5c29b4f956d532781d2b86e4e30f83e6b7878dce18c919451d6ba707c90","LICENSE-MIT":"090a294a492ab2f41388252312a65cf2f0e423330b721a68c6665ac64766753b","README.md":"e934d909a97ddbde3bf7c1e9350895623c3c237fed3f33ccc10261e4517b3d49","clippy.toml":"cba24bf1199526b940d8fe1df76bc10a3c71b404969a702ff18e0b73d90680c0","deny.toml":"b7e686ed2f16b1c84a183dcc4fa9795c67be96a9f8e54db73b95ae58d4b76479","examples/eval.rs":"e50e4f4e6d9d0a774acde0b8dff8e2959f521091ba2d7429bab9dc81c1a5fd0a","release.toml":"42df748d28abd726fa155a29102da0c7ac006dfb601f6bdebfc169049b0ca535","src/error.rs":"f7b5ac813cdaee0d6d021619caff52710918365e13ceb40bbc15f2e526aaf61e","src/expr.rs":"944b8ab027c4dcaa2f7ff7f1df6f23582bad5810af55049ce43bf404700b3059","src/expr/lexer.rs":"95fcdf902827fcccbc94af6052550b5070bc167c1d9c0b9bd5fb598f2b3380a3","src/expr/parser.rs":"4c5864d46b8bb023bd45b4eb2579f0c591b61f5f88d1f16b8e1e0fd4d00095c5","src/lib.rs":"68e82be4c5b8e454fbb3c524b6cc8b9534950bc4f99d9340766eb75405cd9abe","src/targets/builtins.rs":"f078966ea9ec6f5b003664ad36a7598dadb11179188643ae1adceabbaf7893ab","tests/eval.rs":"0c623b4ecd0b0cd69a666954c967f151b35610c8d23ed3f419954f8fef3b2878","tests/lexer.rs":"55109f29f4f3e257e969b53ee7823aeb50212abebbc55d1a67d7393ea42ce80b","tests/parser.rs":"466f77b150fe39eb89db3a157f26431b9367e872630fe213be75f9ffd8177e03"},"package":"0aacacf4d96c24b2ad6eb8ee6df040e4f27b0d0b39a5710c30091baa830485db"}
